### PR TITLE
Transition from Microsoft.Azure.KeyVault to Azure.Security.KeyVault

### PR DESCRIPTION
**Pull request by erdembayar**
Fixes: https://github.com/NuGet/Engineering/issues/4704

Transition from Microsoft.Azure.KeyVault to Azure.Security.KeyVault package dependency. Former one is deprecated.
I manually tested this change in [Gallery dev environment](https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=1478870)

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
   <!-- NuGet dependencies shared across projects -->
   <PropertyGroup>
-    <ServerCommonPackageVersion>2.109.0</ServerCommonPackageVersion>
+    <ServerCommonPackageVersion>2.110.0-agr-kv-lib-upgrade3-8363300</ServerCommonPackageVersion>
```
Comment (erdembayar) [inline Directory.Build.props:5]: TODO: Action update to stable version later

**Comment by joelverhagen** [comment]
There seem to be some unrelated commits in your commit list.

File: src/Ng/CommandHelpers.cs [hunk 2]
```diff
@@ -97,14 +97,19 @@ private static ICachingSecretInjector GetSecretInjector(IDictionary<string, stri
                 }
                 else
                 {
+                    var tenantId = arguments.GetOrThrow<string>(Arguments.TenantId);
                     var clientId = arguments.GetOrThrow<string>(Arguments.ClientId);
```
Comment (joelverhagen) [inline src/Ng/CommandHelpers.cs:101]: What do you think about supporting passing `clientId` in the managed identity path (similar to gallery). As we see from our Managed Identity App Service vs VM conversation, we know it's not necessary, but I think it's good to unlock the same capability (support we ever add system-managed identity or a 2nd user-managed identity).

File: src/NuGet.Jobs.Common/SecretReader/SecretReaderFactory.cs [hunk 3]
```diff
@@ -39,6 +39,7 @@ public ICachingSecretReader CreateSecretReader(IDictionary<string, string> setti
                 keyVaultConfiguration =
                     new KeyVaultConfiguration(
                         JobConfigurationManager.GetArgument(settings, JobArgumentNames.VaultName),
+                        JobConfigurationManager.GetArgument(settings, JobArgumentNames.TenantId),
```
Comment (joelverhagen) [inline src/NuGet.Jobs.Common/SecretReader/SecretReaderFactory.cs:42]: Same as above. What do you think about unlocking `clientId` passing for managed identity?)

File: src/NuGet.Services.AzureSearch/NuGet.Services.AzureSearch.csproj [hunk 4]
```diff
@@ -11,6 +11,7 @@
   <ItemGroup>
     <PackageReference Include="Azure.Identity" />
     <PackageReference Include="Azure.Search.Documents" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime" />
```
Comment (joelverhagen) [inline src/NuGet.Services.AzureSearch/NuGet.Services.AzureSearch.csproj:14]: Why is this needed?

File: Directory.Packages.props [hunk 5]
```diff
@@ -33,6 +33,7 @@
     <PackageVersion Include="Microsoft.Extensions.Primitives" Version="2.2.0" />
     <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
     <PackageVersion Include="Microsoft.SourceLink.GitHub" Version="1.0.0" />
+    <PackageVersion Include="Microsoft.Rest.ClientRuntime" Version="2.3.24" />
```
Comment (joelverhagen) [inline Directory.Packages.props:36]: Why is this dependency needed?

File: src/Ng/CommandHelpers.cs (hunk 2 above)
Comment (erdembayar) [reply src/Ng/CommandHelpers.cs:102]: Sure, I made change in this file. Please check.

File: src/NuGet.Jobs.Common/SecretReader/SecretReaderFactory.cs (hunk 3 above)
Comment (erdembayar) [reply src/NuGet.Jobs.Common/SecretReader/SecretReaderFactory.cs:43]: Sure, I made change in this file. Please check.

File: src/NuGet.Services.AzureSearch/NuGet.Services.AzureSearch.csproj (hunk 4 above)
Comment (erdembayar) [reply src/NuGet.Services.AzureSearch/NuGet.Services.AzureSearch.csproj:14]: It looks like previously this `Microsoft.Rest` was coming as transitive dependency from `Microsoft.Azure.KeyVault`, now we need to reference directly.
image
image

**Review by erdembayar — COMMENTED** [review]
I addressed your comments, please check.

**Review by joelverhagen — APPROVED** [review]
